src/Sylius/Bundle/ShopBundle/EventListener/OrderPromotionIntegrityChecker.php

```diff
@@ -42,15 +49,18 @@
 
     /**
      * @param PromotionEligibilityCheckerInterface $promotionEligibilityChecker
+     * @param PromotionApplicatorInterface $promotionApplicator
      * @param EventDispatcherInterface $eventDispatcher
      * @param RouterInterface $router
      */
     public function __construct(
         PromotionEligibilityCheckerInterface $promotionEligibilityChecker,
+        PromotionApplicatorInterface $promotionApplicator,
```

Comment:
🚨 BC Break

Reply:
The solution here is to add it as the last argument, make it nullable, do `@trigger_error('deprecation message', \E_USER_DEPRECATED)` if that argument is null and do not run added logic if this promotion applicator is not there.

Reply:
@pamil Ok I'll make that change but do not now well what message to put. My english is not so well but if you give me a message suggestion I'll make the change.